Menu: separation from button

agliga opened the pull request:
Split from dropdown.less and used mixin

<!--  Delete any sections below that are not relevant to this PR -->

## Description
<!--- What are the changes? -->
Split up menu to be independent of menu-button. For now, some of the code has been copied over, but ideally would be best to create a common place with all menu items (maybe a menu mixin).
Did not want to add too much more since this is already a large PR.
For this one, I tried to keep as much from dropdown.less into the menu.less, while removing the ones redundant from dropdown-mixins

## Context
<!--- Why did you make these changes, and why in this particular way? -->
I kept fake-menu as a separate item from fame-menu-button. If we want to remove that, just let me know.
I also renamed all menu CSS selectors to menu-button, but I kept the items/item selectors as is.
One caveat, in the dropdown-mixins.less there's a selector for active and .listbox__status. This causes duplicate styles in CSS in combobox and in menu. I added my styles there too in order to match the selectors. However, it added those styles in combobox (as well as the combobox styles in menu). Might consider a minor refactor or some common classes. 

## References
<!-- Include links to JIRA, Github, etc. if appropriate -->
#630 

## Screenshots
<!-- Upload screenshots if appropriate-->

ianmcburnie commented on dist/listbox/ds4/listbox.css:
```diff
@@ -46,6 +46,9 @@ div.listbox__option[role="option"]:hover {
 div.listbox__option[role="option"]:active {
   font-weight: normal;
 }
+div.listbox__option[role="option"]:active .menu-button__status,
```
I think something has gone wrong here. We shouldn't be seeing menu styles in the listbox.css.

agliga replied:
Yes I mentioned that in description. 
The mixin for dropdown has nested .listbox__status styles which will be pulled both in listbox and menu. I added menu-button status styles too in order to follow that, but it creates these extra styles. 
What should probably be done is create a generic status style to apply the styles from mixin to. Otherwise to move those styles in their respective classes

ianmcburnie replied:
Okay, we should try and fix this. For now let's go the simplest, less disruptive route which is probably to just move the styles into their respective classes (as you suggest).

agliga commented:
* Split up the listbox status styles and put them in their own respective classes (I added them as different selectors because lint was not letting me pass with nested selectors)
* Removed the box shadow (split up dropdown so I can get the max-height class, although, is that necessary in the non overlay menu?)
* Put a <div role="menu"> element in all menu-button's

seangates commented on docs/_includes/common/menu.html:
```diff
@@ -122,431 +98,4 @@ <h3 id="menu-radio">Radio Menu</h3>
-    </div>
-    {% endif %}
-</div>
+</div>
```
Nit: need a newline at end of the document.

seangates commented on docs/_includes/common/menu.html:
```diff
@@ -1,7 +1,7 @@
     <h2><span class="secondary-text">@ebay/skin/</span>menu</h2>
 
-    <p>A menu follows the principal of <a href="https://en.wikipedia.org/wiki/Progressive_disclosure">progressive disclosure</a>, it keeps secondary commands hidden from view until needed.</p>
+    <p>A menu can ben shown without to allow navigation to other pages</p>
```
"without" what?

seangates replied:
"ben shown"

ianmcburnie commented on src/less/menu/ds6/menu.less:
```diff
@@ -2,5 +2,5 @@
 @import "../../less/ds6/variables.less";
 @import "../../less/ds6/mixins.less";
-@import "../../dropdown/ds6/dropdown.less";
+@import "../../mixins/dropdown/ds4/dropdown-mixins.less";
```
Ruh roh. Just spotted this. `ds4` instead of `ds6`. Will fix with direct commit to 8.0.0 branch.
